### ISSUE TITLE
Does not work in new custom 1.8 builds (?)

### ISSUE THREAD

**Issue by cppchriscpp** [cppchriscpp/TravelPortals#12]
Have gotten a few messages asking for 1.8 support. While I had one concern about what could break with 1.8, (data values) that concern doesn't seem to be the problem. 

One of the reporters provided a stacktrace: 

```
[20:20:21] [Server thread/ERROR]: Could not pass event BlockPlaceEvent to TravelPortals v2.2.7
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:305) ~[spigot.jar:git-Spigot-eb39b47-0899683]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-eb39b47-0899683]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-eb39b47-0899683]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-eb39b47-0899683]
	at org.bukkit.craftbukkit.v1_8_R1.event.CraftEventFactory.callBlockPlaceEvent(CraftEventFactory.java:125) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.ItemStack.placeItem(ItemStack.java:154) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.PlayerInteractManager.interact(PlayerInteractManager.java:503) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.PlayerConnection.a(PlayerConnection.java:724) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:50) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:80) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.PacketHandleTask.run(SourceFile:13) [spigot.jar:git-Spigot-eb39b47-0899683]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471) [?:1.7.0_65]
	at java.util.concurrent.FutureTask.run(FutureTask.java:262) [?:1.7.0_65]
	at net.minecraft.server.v1_8_R1.MinecraftServer.z(MinecraftServer.java:683) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.DedicatedServer.z(DedicatedServer.java:316) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.MinecraftServer.y(MinecraftServer.java:623) [spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.MinecraftServer.run(MinecraftServer.java:526) [spigot.jar:git-Spigot-eb39b47-0899683]
	at java.lang.Thread.run(Thread.java:745) [?:1.7.0_65]
Caused by: java.lang.IllegalArgumentException: Cannot set property BlockStateInteger{name=level, clazz=class java.lang.Integer, values=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]} to 16 on block minecraft:flowing_water, it is not an allowed value
	at net.minecraft.server.v1_8_R1.BlockData.set(SourceFile:124) ~[spigot.jar:git-Spigot-eb39b47-0899683]
	at net.minecraft.server.v1_8_R1.BlockFluids.fromLegacyData(SourceFile:316) ~[spigot.jar:git-Spigot-eb39b47-0899683]
	at org.bukkit.craftbukkit.v1_8_R1.block.CraftBlock.setData(CraftBlock.java:107) ~[spigot.jar:git-Spigot-eb39b47-0899683]
	at org.bukkit.craftbukkit.v1_8_R1.block.CraftBlock.setData(CraftBlock.java:92) ~[spigot.jar:git-Spigot-eb39b47-0899683]
	at net.cpprograms.minecraft.TravelPortals.TravelPortalsBlockListener.onBlockPlace(TravelPortalsBlockListener.java:121) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor267.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.7.0_65]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[?:1.7.0_65]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:301) ~[spigot.jar:git-Spigot-eb39b47-0899683]
	... 17 more
```

Got another report by email that only one block of water is created, and the structure does not work. Seems likely to be related, but gave no details on any errors or version. Will follow up on that.

Appears to be a problem with Spigot version Spigot-eb39b47-0899683. From the look of the error, I suspect it is a spigot/cb bug, but will confirm/look for workarounds.

**Comment by cppchriscpp** [comment]
Got another report of this today. Definitely seems like a huge problem. Have an idea for a workaround, and should be able to prove the problem so that I can file a bug report with Spigot.

Another affected version: Spigot v1.8.0 R0.1 (Build #20150101)

**Comment by cppchriscpp** [comment]
Turns out the error actually explained exactly what was wrong, and I look a bit dumb. When I first wrote this, it looks like I set the data value to be 16 explicitly, and that rolled over to 0. I've changed it over to explicitly do that and tested against the latest Spigot 1.8 releases. (And also feel quite dumb at this moment) 

A new release has been rolled out and is available now here: http://cpprograms.net/files/TravelPortals.zip

I'll be making the jar available on dev.bukkit.org shortly! Thanks for bearing with me.
